src/server/svr_jobfunc.c

```diff
@@ -7108,14 +7085,81 @@ recreate_exec_vnode(job *pjob, char *vnodelist, char *err_msg,
 	if (deallocated_execvnode[entry] == '+')
 		deallocated_execvnode[entry] = '\0';
 
+	if (deallocated_execvnode[0] != '\0') {
```

Comment:
Moved up these series of calls so that if something fails in them, non of the  job attributes (exec_vnode, exec_host, etc...) would have been set yet.